### ISSUE TITLE
md-select: search does not work when you type the first letter of any other language

### ISSUE THREAD

**Issue by mihrich** [angular/material#7730]
Hello Boys!

The specialist said that the problem could be [here](https://github.com/angular/material/blob/3524aa9573fe29e53341c60ce6127fc7ad323446/src/components/select/select.js):

`var searchStr = '';
var clearSearchTimeout, optNodes, optText;
var CLEAR_SEARCH_AFTER = 300;
self.optNodeForKeyboardSearch = function(e) {
  clearSearchTimeout && clearTimeout(clearSearchTimeout);
  clearSearchTimeout = setTimeout(function() {
    clearSearchTimeout = undefined;
    searchStr = '';
    optText = undefined;
    optNodes = undefined;
  }, CLEAR_SEARCH_AFTER);
  searchStr += String.fromCharCode(e.keyCode);
  var search = new RegExp('^' + searchStr, 'i');
  if (!optNodes) {
    optNodes = $element.find('md-option');
    optText = new Array(optNodes.length);
    angular.forEach(optNodes, function(el, i) {
      optText[i] = el.textContent.trim();
    });
  }
  for (var i = 0; i < optText.length; ++i) {
    if (search.test(optText[i])) {
      return optNodes[i];
    }
  }
};`

Also, experts say that the wrong key code is converted into a symbol.

[Example](http://codepen.io/yeoman/pen/MymmOg?editors=1111)

Version angular is 1.4.8.
Version angular materials is 1.1.0-rc1.

Thank you!

**Comment by clshortfuse** [comment]
@topherfangio I believe we can close this now

**Comment by topherfangio** [comment]
@clshortfuse No, the associated PR still needs some work and has not been merged.
